pkg/materialize/cluster.go

```diff
@@ -191,57 +210,32 @@ func (b *ClusterBuilder) Drop() error {
 	return b.ddl.drop(qn)
 }
 
-func (b *ClusterBuilder) Resize(newSize string) error {
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (SIZE '%s');`, b.QualifiedName(), newSize)
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetSize(newSize string) {
+	b.size = newSize
 }
 
-func (b *ClusterBuilder) SetDisk(disk bool) error {
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (DISK %t);`, b.QualifiedName(), disk)
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetDisk(disk bool) {
+	b.disk = disk
 }
 
-func (b *ClusterBuilder) SetReplicationFactor(newReplicationFactor int) error {
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (REPLICATION FACTOR %d);`, b.QualifiedName(), newReplicationFactor)
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetReplicationFactor(newReplicationFactor int) {
+	b.replicationFactor = &newReplicationFactor
 }
 
-func (b *ClusterBuilder) SetAvailabilityZones(availabilityZones []string) error {
-	az := strings.Join(availabilityZones[:], ",")
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (AVAILABILITY ZONES = [%s]);`, b.QualifiedName(), az)
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetAvailabilityZones(availabilityZones []string) {
+	b.availabilityZones = availabilityZones
 }
 
-func (b *ClusterBuilder) SetIntrospectionInterval(introspectionInterval string) error {
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (INTROSPECTION INTERVAL %s);`, b.QualifiedName(), QuoteString(introspectionInterval))
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetIntrospectionInterval(introspectionInterval string) {
+	b.introspectionInterval = introspectionInterval
 }
 
-func (b *ClusterBuilder) SetIntrospectionDebugging(introspectionDebugging bool) error {
-	q := fmt.Sprintf(`ALTER CLUSTER %s SET (INTROSPECTION DEBUGGING %t);`, b.QualifiedName(), introspectionDebugging)
-	return b.ddl.exec(q)
+func (b *ClusterBuilder) SetIntrospectionDebugging(introspectionDebugging bool) {
+	b.introspectionDebugging = introspectionDebugging
 }
 
-func (b *ClusterBuilder) SetSchedulingConfig(s interface{}) error {
-	schedulingConfig := GetSchedulingConfig(s)
-	var scheduleStatement string
-	var q string
-
-	// Check if the scheduling is enabled and set the appropriate SQL command.
-	if schedulingConfig.OnRefresh.Enabled {
-		scheduleStatement = "SCHEDULE = ON REFRESH"
-		if schedulingConfig.OnRefresh.HydrationTimeEstimate != "" {
-			scheduleStatement += fmt.Sprintf(" (HYDRATION TIME ESTIMATE = %s)", QuoteString(schedulingConfig.OnRefresh.HydrationTimeEstimate))
-		} else if schedulingConfig.OnRefresh.RehydrationTimeEstimate != "" {
-			scheduleStatement += fmt.Sprintf(" (HYDRATION TIME ESTIMATE = %s)", QuoteString(schedulingConfig.OnRefresh.RehydrationTimeEstimate))
-		}
-		q = fmt.Sprintf("ALTER CLUSTER %s SET (%s);", b.QualifiedName(), scheduleStatement)
-	} else {
-		// Reset the schedule settings if not enabled.
-		q = fmt.Sprintf("ALTER CLUSTER %s RESET (SCHEDULE);", b.QualifiedName())
-	}
```

Comment:
This looks great! Thanks a lot @jubrad!

The only issue that I've noticed while manually testing this is that if you were to change the scheduling from enabled true to false, it generates the following alter statement which will fail:

```sql
ALTER CLUSTER "scheduling_cluster" SET ();
```

Repro:

```hcl
resource "materialize_cluster" "scheduling_cluster" {
  name = "scheduling_cluster"
  size = "50cc"
  scheduling {
    on_refresh {
      enabled                 = true # Apply and then change to false and apply again
      hydration_time_estimate = "3 hour"
    }
  }
}
```